Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -37,29 +38,79 @@ public class RazorTargetTests
 
         public TestContext TestContext { get; set; }
 
+        [TestMethod]
+        public void Razor_RazorSpecificOutputAndProjectInfo_AreCopiedToCorrectFolders()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSpecificOutDir = Path.Combine(root, "0");
+            var temporaryProjectSpecificOutDir = Path.Combine(root, "0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.Razor");
+            Directory.CreateDirectory(projectSpecificOutDir);
+            Directory.CreateDirectory(temporaryProjectSpecificOutDir);
+            File.WriteAllText(Path.Combine(projectSpecificOutDir, "ProjectInfo.xml"),
+$@"
+<ProjectInfo>
+  <AnalysisSettings>
+    <Property Name = ""sonar.cs.roslyn.reportFilePaths"" > C:\SonarSource\Sandbox\Mixed\Mixed\Main2\bin\Debug\netcoreapp3.1\Main2.dll.RoslynCA.json | C:\SonarSource\Sandbox\Mixed\Mixed\Main2\bin\Debug\netcoreapp3.1\Main2.dll.RoslynCA.json </ Property >
+    <Property Name = ""sonar.cs.analyzer.projectOutPaths"" >{projectSpecificOutDir}</Property>
+  </ AnalysisSettings >
+  <Configuration > Debug </ Configuration >
+  <Platform > AnyCPU </ Platform >
+  <TargetFramework > netcoreapp3.1 </TargetFramework>
+</ProjectInfo> ");
+            File.WriteAllText(Path.Combine(temporaryProjectSpecificOutDir, "ProtoBuf.txt"), string.Empty);
+            var razorSpecificProjectInfo = Path.Combine(razorSpecificOutDir, "ProjectInfo.xml");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <SonarTemporaryProjectSpecificOutDir>{temporaryProjectSpecificOutDir}</SonarTemporaryProjectSpecificOutDir>
+  <ProjectSpecificOutDir>{projectSpecificOutDir}</ProjectSpecificOutDir>
+</PropertyGroup>
+";
+
+            var filePath = CreateProjectFile(null, projectSnippet, TargetConstants.RazorSonarCopyProjectInfoFile);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.RazorSonarCopyProjectInfoFile);
+
+            // Assert
+            result.AssertTargetExecuted(TargetConstants.RazorSonarCopyProjectInfoFile);
+            File.Exists(Path.Combine(projectSpecificOutDir, "ProtoBuf.txt")).Should().BeTrue();
+            File.Exists(razorSpecificProjectInfo).Should().BeTrue();
+            File.ReadAllText(razorSpecificProjectInfo).Should().Contain($@"<Property Name = ""sonar.cs.analyzer.projectOutPaths"" >{razorSpecificOutDir}</Property>");
```

Comment:
This should be doable in a object oriented clean way with `ProjectInfoAssertions.AssertProjectInfoExists` that will return deserialized `ProjectInfo` object that you can properly assert.